etna/models/base.py

```diff
@@ -450,4 +467,303 @@ def get_model(self) -> Any:
+        ts:
+            Samples with data to fit on.
+
+        Returns
+        -------
+        :
+            Trained Model
+        """
+        pass
+
+    @abstractmethod
+    def raw_predict(self, ts: "Dataset") -> Dict[Tuple[str, str], np.ndarray]:
+        """Make inference on torch like Dataset.
+
+        Parameters
+        ----------
+        ts:
+            Samples with data to make inference on.
+
+        Returns
+        -------
+        :
+            Predictions for each segment
+        """
+        pass
+
+    @abstractmethod
+    def step(self, batch: dict, *args, **kwargs) -> Tuple["torch.Tensor", "torch.Tensor", "torch.Tensor"]:
+        """Make batch step.
+
+        Parameters
+        ----------
+        batch:
+            Batch with data to make inference on.
+
+        Returns
+        -------
+        :
+            loss, true_target, prediction_target
+        """
+        pass
+
+
+class DeepBaseModel(LightningModule, FitAbstractModel, DeepBaseAbstractModel, BaseMixin):
+    """Class for partially implemented interfaces."""
+
+    def __init__(
+        self,
+        *,
+        encoder_length: int,
+        decoder_length: int,
+        train_batch_size: int,
+        test_batch_size: int,
+        train_dataloader_kwargs: dict,
+        test_dataloader_kwargs: dict,
+        val_dataloader_kwargs: dict,
+        split_kwargs: dict,
+        trainer_kwargs: dict,
+    ):
+        """Init DeepBaseModel.
+
+        Parameters
+        ----------
+        encoder_length:
+            encoder length
+        decoder_length:
+            decoder length
+        train_batch_size:
+            batch size for training
+        test_batch_size:
+            batch size for testing
+        train_dataloader_kwargs:
+            kwargs for train dataloader like sampler for example
+        test_dataloader_kwargs:
+            kwargs for test dataloader
+        val_dataloader_kwargs:
+            kwargs for validation dataloader
+        split_kwargs:
+            kwargs for torch dataset split for train-test splitting
+        trainer_kwargs:
+            Pytorch ligthning trainer kwargs
+        """
+        super().__init__()
+        self.encoder_length = encoder_length
+        self.decoder_length = decoder_length
+        self.train_batch_size = train_batch_size
+        self.test_batch_size = test_batch_size
+        self.trainer_kwargs = trainer_kwargs
+        self.train_dataloader_kwargs = train_dataloader_kwargs
+        self.val_dataloader_kwargs = val_dataloader_kwargs
+        self.test_dataloader_kwargs = test_dataloader_kwargs
+        self.split_kwargs = split_kwargs
+
+    @log_decorator
+    def fit(self, ts: TSDataset) -> "DeepBaseModel":
+        """Fit Model.
+
+        Parameters
+        ----------
+        ts:
+            TSDataset with features
+
+        Returns
+        -------
+        :
+            Model after fit
+        """
+        ts_torch = ts.to_torch_dataset(self.make_samples)
```

Comment:
Do we need to pass `drop_na` parameter somehow here?

Reply:
add drop_na explicitly